### ISSUE TITLE
llama-funasr-sensevoice intermittently returns empty transcript on valid speech (exit 0, no error)

### ISSUE THREAD

**Issue by emumanx** [huaxin0/FunASR-GGML#4]
**Title:** llama-funasr-sensevoice intermittently returns empty transcript on valid speech (exit 0, no error)

**Describe the bug**
`llama-funasr-sensevoice` occasionally produces an **empty transcript** on valid, clearly-audible speech. The process exits 0, prints `[sensevoice] done 0.11s`, but emits no text. Re-running the exact same audio file on the same binary/model returns the correct transcript, so it is intermittent, not deterministic.

**To reproduce**
Run the binary on a short (~1-4s) WAV of clear speech, repeatedly:
```
llama-funasr-sensevoice -m sensevoice-small-f16.gguf -a input.wav --keep-tags
```
Most runs return the transcript; occasionally (roughly 1 in several) it prints only `[sensevoice] done 0.11s` with no output. The same file that failed will transcribe correctly on a subsequent run.

**Expected behavior**
Every run on valid speech should return the transcript.

**Actual behavior**
Intermittently returns empty output with exit code 0 and no error message.

**Environment**
- Binary: `llama-funasr-sensevoice` (from the prebuilt `funasr-llamacpp-linux-x64-avx2.tar.gz`, ~2.6MB)
- Model: `sensevoice-small-f16.gguf` (470MB, from FunAudioLLM/SenseVoiceSmall-GGUF)
- OS: Linux (Ubuntu 25.10), x86_64 AVX2
- CPU-only, no GPU

**Additional context**
- The audio is confirmed valid (56% of samples above speech threshold, RMS ~1500, peak ~8700).
- The failure is transient: the identical file transcribes correctly on retry.
- This causes dropped transcriptions in a voice-input pipeline; a retry-on-empty works around it but the root cause is in the binary.

**Comment by LauraGPT** [comment]
Thanks for the precise report. I reproduced this with the exact `runtime-llamacpp-v0.2.3` Linux AVX2 binary and the published `sensevoice-small-f16.gguf`.

Root cause: the F16 GGUF stores `embed.weight` as F16 `[560, 16]`, but the runtime reinterpreted that storage as `float*`. Query-token copies therefore read past the tensor boundary. The resulting CTC logits depended on adjacent process memory, which explains why identical audio could sometimes collapse to blank tokens while the compute call and process still returned success.

Evidence:
- Before the fix: 22 newline-only outputs among 298 completed sequential runs; every compute status was 0.
- After the fix: 100/100 runs produced one identical non-empty output.
- The corrected F16 output matches the Q8 model (whose query embedding is F32) byte-for-byte.

Fix PR: https://github.com/modelscope/FunASR/pull/3550

Until a fixed runtime archive is published, `sensevoice-small-q8.gguf` is a safe workaround for this specific bug because its `embed.weight` is F32. Retrying empty F16 results also avoids dropped utterances, but it is only a workaround. I will follow up here when the fixed release asset is available.
